aiohomekit/controller/ip/pairing.py

```diff
@@ -280,28 +276,36 @@ async def put_characteristics(self, characteristics):
         if not self.accessories:
             await self.list_accessories_and_characteristics()
 
-        data = []
-        characteristics_set = set()
+        char_payload: list[dict[str, Any]] = []
+        listener_update: dict[tuple[int, int], dict[str, Any]] = {}
         for characteristic in characteristics:
-            aid = characteristic[0]
-            iid = characteristic[1]
-            value = characteristic[2]
-            characteristics_set.add(f"{aid}.{iid}")
-            data.append({"aid": aid, "iid": iid, "value": value})
-        data = {"characteristics": data}
-
-        response = await self.connection.put_json("/characteristics", data)
+            aid, iid, value = characteristic
+            char_payload.append({"aid": aid, "iid": iid, "value": value})
+            accessory_chars = self.accessories.aid(aid).characteristics
```

Comment:
This does a linear search under the hood which could be an issue with many acessories.  We should optimize that

Reply:
https://github.com/Jc2k/aiohomekit/pull/248